thentos-adhocracy/tests/Thentos/Adhocracy3/Backend/Api/SimpleSpec.hs

```diff
@@ -91,6 +95,40 @@ spec =
                      (Aeson.eitherDecode userdata :: Either String A3UserWithPass)
                      `shouldBe` Left "Illegal whitespace sequence in user name: \" Anna  Toll\""
 
+        describe "login" . before setupTestBackend . after teardownTestBackend $
+            it "rejects bad credentials mimicking A3" $
+                \bts -> runTestBackend bts $ do
+
+                    let reqBody = encodePretty . object $
+                            [ "name"     .= String "Anna Müller"
+                            , "password" .= String "this-is-wrong"
+                            ]
+
+                    rsp <- srequest $ makeSRequest "POST" "login_username" [] reqBody
+                    shouldBeErr400WithCustomMessage (simpleStatus rsp) (simpleBody rsp)
+                        "\"User doesn't exist or password is wrong\""
+
+        describe "activate_account" . before setupTestBackend . after teardownTestBackend $
+            it "rejects bad path mimicking A3" $
+                \bts -> runTestBackend bts $ do
+
+                    let reqBody = encodePretty . object $
+                            [ "path" .= String "/activate/no-such-path" ]
+
+                    rsp <- srequest $ makeSRequest "POST" "activate_account" [] reqBody
```

Comment:
I'm not sure what you mean with the `[json|...]` in the last line but I would suppose that would result in a bit-for-bit comparison that the encoded JSON matches the response body? I feel that would be testing too much---the same JSON can be encoded in various ways, so we shouldn't expect that all fields are written in the same order etc.

Also I think that for each `it` we would have to do a separate `runTestBackend bts`, so not only would that actually be quite long-ish, but having too many separate `it`s would also slow things down.

I know that some people swear to the style "test just *one thing* per test" and I agree that that's reasonable in theory, but considering that our test suite is already quite slow I think we shouldn't be too pedantic about that in practice.